api/types/server_test.go

```diff
@@ -354,6 +357,268 @@ func TestServerCheckAndSetDefaults(t *testing.T) {
 				require.EqualError(t, err, `invalid SubKind "invalid-subkind"`)
 			},
 		},
+		{
+			name: "OpenSSHEphemeralKey node without cloud metadata",
+			server: &ServerV2{
+				Kind:    KindNode,
+				SubKind: SubKindOpenSSHEphemeralKeyNode,
+				Version: V2,
+				Metadata: Metadata{
+					Name:      "5da56852-2adb-4540-a37c-80790203f6a9",
+					Namespace: defaults.Namespace,
+				},
+				Spec: ServerSpecV2{
+					Addr:     "example:22",
+					Hostname: "openssh-node",
+				},
+			},
+			assertion: func(t *testing.T, s *ServerV2, err error) {
+				require.EqualError(t, err, `AWS CloudMetadata is required for "openssh-ephemeral-key" SubKind`)
+			},
+		},
+		{
+			name: "OpenSSHEphemeralKey node without cloud metadata but missing aws info",
+			server: &ServerV2{
+				Kind:    KindNode,
+				SubKind: SubKindOpenSSHEphemeralKeyNode,
+				Version: V2,
+				Metadata: Metadata{
+					Name:      "5da56852-2adb-4540-a37c-80790203f6a9",
+					Namespace: defaults.Namespace,
+				},
+				Spec: ServerSpecV2{
+					Addr:          "example:22",
+					Hostname:      "openssh-node",
+					CloudMetadata: &CloudMetadata{},
+				},
+			},
+			assertion: func(t *testing.T, s *ServerV2, err error) {
+				require.EqualError(t, err, `AWS CloudMetadata is required for "openssh-ephemeral-key" SubKind`)
+			},
+		},
+		{
+			name: "OpenSSHEphemeralKey node with aws cloud metadata but missing accountid",
+			server: &ServerV2{
+				Kind:    KindNode,
+				SubKind: SubKindOpenSSHEphemeralKeyNode,
+				Version: V2,
+				Metadata: Metadata{
+					Name:      "5da56852-2adb-4540-a37c-80790203f6a9",
+					Namespace: defaults.Namespace,
+				},
+				Spec: ServerSpecV2{
+					Addr:     "example:22",
+					Hostname: "openssh-node",
+					CloudMetadata: &CloudMetadata{
+						AWS: &AWSInfo{
+							InstanceID:       "i-123456789012",
+							Region:           "us-east-1",
+							AvailabilityZone: "us-east-1b",
+							VPCID:            "vpc-abcd",
+							Integration:      "teleportdev",
+						},
+					},
+				},
+			},
```

Comment:
Suggestion: for tests like these, I like starting with a valid instance and changing it to cause the wanted error.

For example, consider something like this:

```suggestion
			server: makeOpenSSHEmphemeral(func (s *ServerV2) {
				s.Spec.CloudMetadata.AccountID = ""
			}),
```

I'll often declare the equivalent to `makeOpenSSHEmphemeral` in the body of the test itself, as in:

```go
makeOpenSSHEmphemeral = func(fn func(s *ServerV2)) *ServerV2 {
	s := ... // Declare a valid server instance here.
	fn(s)
	return s
}
```

Many ways to do it.